.github/workflows/build-and-run.yml

```
@@ -0,0 +1,109 @@
+name: Build-And-Run Tests
+
+on: [push]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  build_and_test:
+    env:
+      ALL_EXCLUDE: stage_dem|ionosphere|background|pybind.unwrap.phass
+
+    defaults:
+      run:
+        shell: bash -l {0}
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - label: Linux
+            runner: ubuntu-latest
+            ctest_exclude: -E ".*($ALL_EXCLUDE)"
+          - runner: macos-latest
+            label: macOS
+            ctest_exclude: -E ".*($ALL_EXCLUDE|raster|unwrap.snaphu)"
+        deps:
+          - label: Latest
+            env-file: env/environment.yml
+            extra-specs: ""
+        build-type: [ Release, Debug ]
+        include:
+          - os:
+              label: Linux
+              runner: ubuntu-latest
+              ctest_exclude: -E ".*($ALL_EXCLUDE)"
+            deps:
+              label: Minimum
+              env-file: env/minimum-reqs.yml
+              extra-specs: "gsl"
+            build-type: Release
+
+    name: Build And Test - ${{ matrix.os.label }} ${{ matrix.deps.label }} ${{ matrix.build-type }}
+    runs-on: ${{ matrix.os.runner }}
+    steps:
+      # Typical github repo checkout step
+      - name: Github Repo Checkout
+        uses: actions/checkout@v2
+
+      # Prints the variables for this run to simplify debugging
+      - name: Print Run Variables
+        run: |
+          echo Runner OS: ${{ matrix.os.runner }}
+          echo Environment: ${{ matrix.deps.env-file }}
+          echo Build Type: ${{ matrix.build-type }}
+          echo ctest exclude command: ${{ matrix.os.ctest_exclude }}
+          echo Depenencies: ${{ matrix.deps.extra-specs }}
+
+      # Set the conda environment up using Mamba and the dependencies for this run
+      - name: Setup Environment
+        uses: mamba-org/provision-with-micromamba@main
+        with:
+          environment-file: ${{ matrix.deps.env-file }}
+          environment-name: isce3-ci-sandbox
+          extra-specs: ${{ matrix.deps.extra-specs }}
+
+      # Preset environment variables in the conda environment for later steps
+      - name: Environment Variables Preset
+        run: |
+          export ISCE3_PREFIX=${GITHUB_WORKSPACE}/install
+          conda env config vars set ISCE3_PREFIX=$ISCE3_PREFIX
+          conda env config vars set PYTHONPATH=$PYTHONPATH:$ISCE3_PREFIX/packages
+          if test -d "$ISCE3_PREFIX/lib64"; then
+            export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$ISCE3_PREFIX/lib64:$CONDA_PREFIX/bin/ld
+          else
+            export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$ISCE3_PREFIX/lib:$CONDA_PREFIX/bin/ld
+          fi
+          conda env config vars set LD_LIBRARY_PATH=$LD_LIBRARY_PATH
```

Is it necessary to both `export LD_LIBRARY_PATH` \*and\* add `LD_LIBRARY_PATH` to the conda environment using `conda env config vars set`?

I think we could probably skip the first step and just add it to the conda environment

```suggestion
            conda env config vars set LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$ISCE3_PREFIX/lib64
          else
            conda env config vars set LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$ISCE3_PREFIX/lib
          fi
```

---

The reason I'm doing these export-and-then-add-to-environment things is to prevent the runner from having to deactivate and reactivate the environment. `conda env config vars set` requires the environment to be deactivated and reactivated before the variable becomes available for use - so if I need the variable later (e.g. using $ISCE3_PREFIX to set $LD_LIBRARY_PATH and then using $LD_LIBRARY_PATH to set $DYLD_LIBRARY_PATH) I'll have to deactivate the environment and reactivate it which requires some extra steps and and a couple of extra seconds of runtime.

Using `export` first sets the environment variable and makes it available for use within this step and requiring no deactivation and reactivation, and then using `conda env config vars set` makes it available for use in the following steps of the workflow. I've found this to be a faster and simpler way of doing the job.

Another alternative is also for me to just set these variables without referencing each other at all, which I believe would require an additional if/else statement and may make the script a little slower to modify but would delete these export/conda env config vars set steps:

`conda env config vars set ISCE3_PREFIX=${GITHUB_WORKSPACE}/install
conda env config vars set PYTHONPATH=$PYTHONPATH:${GITHUB_WORKSPACE}/install/packages
if test -d "${GITHUB_WORKSPACE}/install/lib64"; then
  conda env config vars set LD_LIBRARY_PATH=$LD_LIBRARY_PATH:${GITHUB_WORKSPACE}/install/lib64
  if [ "$RUNNER_OS" == "macOS" ]; then
    conda env config vars set DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:${GITHUB_WORKSPACE}/install/lib64
  fi
else
  conda env config vars set LD_LIBRARY_PATH=$LD_LIBRARY_PATH:${GITHUB_WORKSPACE}/install/lib
  if [ "$RUNNER_OS" == "macOS" ]; then
    conda env config vars set DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:${GITHUB_WORKSPACE}/install/lib
  fi
fi`

---

Scratch this! With the above comment about moving LD_LIBRARY_PATH setting to after `cmake --build build --target install` has been run, I don't end up having much of a reason to do these export statements anyway.

---

Oh right gotcha -- you needed to export `LD_LIBRARY_PATH` as well because you were using it to populate `DYLD_LIBRARY_PATH` as well in that same workflow step. Sorry, I forgot about that part when I made the initial comment but it makes sense now.